### PR TITLE
Rework integration tests

### PR DESCRIPTION
**Pull request by rictic**
They now:

* Pass locally
* Pass on travis
* Are easy to read
* Are fixture-and-golden based

TODO:

* They need to be much faster. I plan to fix this by also making regular runs of wct much faster.
* The goldens are a bit hard to write, we should support generating them.

This PR is based largly from extracting the testing changes from #374, along with debugging a
few remaining subtleties.

 - [x] CHANGELOG.md has not been updated, as this change is entirely internal, and has no user-visible aspects.

### CODE REVIEW

**Comment by azakus** [comment]
Hush googlebot

File: test/integration/browser.ts [hunk 1]
```diff
@@ -24,360 +26,129 @@ import {Context} from '../../runner/context';
...
+    [testName: string]: [string, string];
+  };
 }
 
-function assertFailed(context: TestContext, expectedError: string) {
-  expect(context.runError).to.eq(expectedError);
-  expect(context.testRunnerError).to.be.eq(expectedError);
-  expect(context.errors).to.deep.equal(repeatBrowsers(context, expectedError));
-}
+type Golden = VariantsGolden|VariantResultGolden;
```
Comment (justinfagnani) [inline test/integration/browser.ts]: is variants support part of getting integration tests working?

File: test/integration/browser.ts [hunk 2]
```diff
@@ -24,360 +26,129 @@ import {Context} from '../../runner/context';
...
-  });
-
-  runsIntegrationSuite('many-js', function(testContext) {
-
-    it('passes', function() {
-      assertPassed(testContext);
-      assertStats(testContext, 6, 0, 0, 'complete');
-    });
+function runsAllIntegrationSuites() {
+  let integrationDirnames = fs.readdirSync(integrationDir);
```
Comment (justinfagnani) [inline test/integration/browser.ts]: const

**Comment by justinfagnani** [comment]
Looks like a lot of variant-related things in there... are those just names?

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Update .npmignore whenever you update this file!
 
 .todo
-bower_components
+/bower_components
 node_modules
```
Comment (rictic) [inline .gitignore]: Yeah. Added for node_modules 'cause that'll come up in a future PR no doubt.

File: gulpfile.js [hunk 4]
```diff
@@ -188,5 +186,5 @@ function commonDepCheck(options) {
 }
 
 gulp.task('prepublish', function(done) {
-  runSequence('build-all', 'test', done);
+  runSequence('build-all', 'test:unit', done);
```
Comment (rictic) [inline gulpfile.js]: No, we don't want to do this in prepublish at all because `npm install` runs prepublish. Added a comment.

**Comment by rictic** [comment]
Rebased out the merge commit that googlebot was unhappy about

**Comment by justinfagnani** [comment]
Hmm, failure of one of the fixtures on node 6

**Comment by rictic** [comment]
The tests are still somewhat flaky, though it seems moreso on travis.

**Pull request merged**